widget/label.go

```diff
@@ -162,3 +177,36 @@ func (l *Label) updateFromData(data binding.DataItem) {
 	}
 	l.SetText(val)
 }
+
+type labelRenderer struct {
+	l *Label
+}
+
+func (r *labelRenderer) Destroy() {
+}
+
+func (r *labelRenderer) Layout(s fyne.Size) {
+	r.l.selection.Resize(s)
+	r.l.provider.Resize(s)
+}
+
+func (r *labelRenderer) MinSize() fyne.Size {
+	return r.l.provider.MinSize()
+}
+
+func (r *labelRenderer) Objects() []fyne.CanvasObject {
+	if !r.l.Selectable {
+		return []fyne.CanvasObject{r.l.provider}
+	}
+
+	return []fyne.CanvasObject{r.l.selection, r.l.provider}
+}
+
+func (r *labelRenderer) Refresh() {
+	r.l.provider.Refresh()
+
+	sel := r.l.selection
```

Comment:
I think we could do the same optimization here to avoid refreshing sel if the label is not selectable?